spec/features/posts_index_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :feature do
+  before :each do
+    @user1 = User.create!(name: 'Abu', photo: 'https://unsplash.com/photos/1.jpg', bio: 'Teacher from Mexico.')
+    @post1 = Post.create!(author: @user1, title: 'First Post', text: 'First text')
+    @post2 = Post.create!(author: @user1, title: 'Second Post', text: 'Second text')
+    @comment1 = Comment.create!(post: @post1, user: @user1, text: 'Hi Abu!')
+    @comment2 = Comment.create!(post: @post1, user: @user1, text: 'Hi Abu!')
+    @like1 = Like.create!(post: @post1, user: @user1)
+    @like2 = Like.create!(post: @post1, user: @user1)
+  end
+  describe 'Index page' do
+    it "should display the user's profile picture" do
+      visit user_posts_path(@user1)
+      expect(page).to have_css("img[src*='https://unsplash.com/photos/1.jpg']")
+    end
+
+    it "should display the user's username" do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@user1.name)
+    end
+
+    it "should display the user's number of posts" do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@user1.post_counter)
+    end
+
+    it "should display the post's title" do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post1.title)
+    end
+
+    it "should display some of the post's body" do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post1.text)
+    end
+
+    it 'should display the first comments on a post' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@comment1.text)
+      expect(page).to have_content(@comment2.text)
+    end
+
+    it 'should display the number of comments on a post' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post1.comments_counter)
+    end
+
+    it 'should display the number of likes on a post' do
+      visit user_posts_path(@user1)
+      expect(page).to have_content(@post1.likes_counter)
+    end
+
+    it "should redirect me to that post's show page when clicking on the post's title" do
+      visit user_posts_path(@user1)
+      click_link @post1.title
+      expect(page).to have_current_path(user_post_path(@user1, @post1))
+    end
+  end
```

Comment:
- [ ] @Team Great work adding tests for the user index page, however, its missing the test highlighted below listed below
![image](https://github.com/AbuTalha3/my_blog_rails/assets/61408860/9d9cac16-393f-41e1-9a86-f4502879644c)

- [ ] @Team Kindly check on some of your tests they are failing as shown below:
![image](https://github.com/AbuTalha3/my_blog_rails/assets/61408860/b0b93a2c-636c-4cf5-91da-b99e385dca40)